content/en/docs/concepts/cluster-administration/system-logs.md

```diff
@@ -0,0 +1,117 @@
+---
+reviewers:
+- dims
+- 44past4
+title: System Logs
+content_type: concept
+weight: 60
+---
+
+<!-- overview -->
+
+System component logs record things happening in cluster, which can be very useful for debugging.
+You can configure log verbosity to see more or less detail.
+Logs can be as coarse-grained as showing errors within a component, or as fine-grained as showing step-by-step traces of events (like HTTP access logs, pod state changes, controller actions, or scheduler decisions).
+
+<!-- body -->
+
+## Klog
+
+Kubernetes components use the [klog](https://github.com/kubernetes/klog) logging library.
+[klog](https://github.com/kubernetes/klog) configures logging via command line flags exposed by components.
+You can find their list in [command-line-tools-reference]. Example of klog format:
+
+```
+I1025 00:15:15.525108       1 httplog.go:79] GET /api/v1/namespaces/kube-system/pods/metrics-server-v0.3.1-57c75779f-9p8wg: (1.512ms) 200 [pod_nanny/v0.0.0 (linux/amd64) kubernetes/$Format 10.56.1.19:51756]
+```
+
+[command line tools reference]: /docs/reference/command-line-tools-reference
+[klog]: https://github.com/kubernetes/klog
+
+### Structured Logging
+
+{{<warning>}}
+Migration to structured log messages is an ongoing process. Not all log messages are structured in this version. When parsing log files, you must also handle unstructured log messages.
+
+Structured logging is an alpha feature. Log formatting and value serialization are subject to change.
+{{< /warning>}}
+
+Structured logging is a effort to introduce a uniform structure in log messages allowing for easy extraction of information, making logs easier and cheaper to store and process.
+New message format is backward compatible and enabled by default.
+
+Format of structured logs:
+```
+<klog header> "<message>" <key1>="<value1>" <key2>="<value2>" ...
+```
+
+Example:
+```
+I1025 00:15:15.525108       1 controller_utils.go:116] "Pod status updated" pod="kube-system/kubedns" status="ready"
+```
+
+
+### JSON log format
+
+The `--logging-format=json` flag changes format of logs from components to JSON format.
+
+{{<warning >}}
+JSON output does not support many standard klog flags. The [command line tools reference] lists unsupported klog flags.
```

Comment:
nit: I have not checked. How do the individual tool components list the `unsupported` klog flags?

Reply:
List of unsupported klog flags are provided as part of flag description. Flag description is available as part of --help, and on https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/

From --help:
```
Logs flags:

      --logging-format string                                                                                                                                        
                Sets the log format. Permitted formats: "text", "json".
                Non-default formats don't honor these flags: --add_dir_header, --alsologtostderr, --log_backtrace_at, --log_dir, --log_file, --log_file_max_size,
                --logtostderr, --skip_headers, --skip_log_headers, --stderrthreshold, --log-flush-frequency.
                Non-default choices are currently alpha and subject to change without warning. (default "text")
```